tracker/event_tracker.go

```diff
@@ -64,6 +66,26 @@ func (e *EventTracker) Start(ctx context.Context) error {
 	blockMaxBacklog := e.numBlockConfirmations*2 + 1
 	blockTracker := blocktracker.NewBlockTracker(provider.Eth(), blocktracker.WithBlockMaxBacklog(blockMaxBacklog))
 
+	go func() {
+		<-ctx.Done()
+		blockTracker.Close()
+		store.Close()
+	}()
+
+	// Init and start block tracker concurrently, with infinite retries
+	go common.RetryForever(ctx, time.Second, func(context.Context) error {
```

Comment:
What if block tracker configuration is not populated how it should be? then we will have endless retries

Reply:
> then we will have endless retries

This is desired behaviour because point 2 in description:
```
2. It allows for more reliable alerting and monitoring based on repeating critical errors.
```

Reply:
Critical errors on startup should not be retried unless they can be temporary in nature and doesn't need user interaction to be fixed.

Reply:
Yes but these are concurrent, unhandled errors. The initialization we can make synchronous and error out trivially. However the sync could take a long time and needs to be concurrent. In that case, failure needs to be propagated to the server logic so that we don't have a deployed Edge relayer failing to handle events, blocking deposits.

This PR began to implement concurrent error propagation https://github.com/0xPolygon/polygon-edge/pull/1373 but in the end I thought it was diverging from the current design of Edge.